packages/devtools_app/lib/src/shared/console/widgets/evaluate.dart

```diff
@@ -397,4 +401,33 @@ class ExpressionEvalFieldState extends State<ExpressionEvalField>
       );
     });
   }
+
+  /// If [expressionText] is assignment like `var x=$1`, processes it.
+  ///
+  /// Returns true is the text was parsed as assignment.
+  bool _tryProcessAssignment(String expressionText) {
+    if (!FeatureFlags.evalAndBrowse) return false;
+
+    final assignment = ConsoleVariableAssignment.tryParse(expressionText);
+    if (assignment == null) return false;
+
+    final variable =
+        serviceManager.consoleService.itemAt(assignment.consoleItemIndex + 1);
+    final value = variable?.value;
+    if (value is! InstanceRef) {
+      _emitToConsole(
+        'Item #${assignment.consoleItemIndex} cannot be assigned to variable.',
+      );
+      return true;
+    }
+
+    evalService.scope[assignment.variableName] = value.id!;
+
+    _emitToConsole(
+      'Variable ${assignment.variableName} is created and can be used '
```

Comment:
Nit: consider "Variable ${...} has been created and can now be used in expression evaluation."